### PR TITLE
Added support for GitHub using AWS CodeStarSourceConnection

### PR DESCRIPTION
**Pull request by asifma**
Added parameters for GitHub support using AWS CodeStarSourceConnection
### Feature or Bugfix
- Feature
### Detail
- Added support for using GitHub as source with CodeStar by implementing the AWS CodeStarSourceConnection.
- Replaced the use of PAT Token with a more secure method using CodeStar connections.
- This is the recommended method. See: https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.pipelines/CodePipelineSource.html
### Relates
- Link to the CodeStar connection setup documentation: [CodeStar Connection Documentation](https://docs.aws.amazon.com/dtconsole/latest/userguide/connections-create-github.html)
- Link to the relevant documentation update that should be updated by someone: [AWS DataAll Deployment Documentation](https://awslabs.github.io/aws-dataall/deploy-aws/)
### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).
1. **Does this PR introduce or modify any input fields or queries - this includes fetching data from storage outside the application (e.g. a database, an S3 bucket)?**
   - Yes, this PR introduces support for inputting CodeStarSourceConnection ARN.
2. **Is the input sanitized?**
   - Yes, the input is properly sanitized before being processed.
3. **What precautions are you taking before deserializing the data you consume?**
   - We ensure that data is properly validated and sanitized before any deserialization takes place.
4. **Is injection prevented by parameterizing queries?**
   - Injection is prevented by utilizing parameterized queries.
5. **Have you ensured no `eval` or similar functions are used?**
   - We have ensured that no `eval` or similar functions are used.
6. **Does this PR introduce any functionality or component that requires authorization?**
   - Yes, this PR introduces a new way to authenticate with GitHub using CodeStar connections.
7. **How have you ensured it respects the existing AuthN/AuthZ mechanisms?**
   - We have ensured that the existing authentication and authorization mechanisms are respected.
8. **Are you logging failed auth attempts?**
   - N/A
9. **Are you using or adding any cryptographic features?**
   - No, this PR does not involve the use or addition of any cryptographic features.
10. **Do you use a standard proven implementation?**
    - Yes
11. **Are the used keys controlled by the customer? Where are they stored?**
    - No keys are used in this context.
12. **Are you introducing any new policies/roles/users?**
    - No new policies, roles, or users are being introduced with this PR.
13. **Have you used the least-privilege principle? How?**
    - We have strictly followed the principle of least privilege in the implementation of this feature.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by louishourcade — APPROVED** [review]
Thanks for your contribution

**Comment by dlpzx** [comment]
Hi @asifma thanks for directly opening a PR :) I have read the docs and I agree with the general purpose of the PR. I will add some comments and we will sync internally to merge it in the following weeks.

File: deploy/app.py [hunk 1]
```diff
@@ -69,6 +69,8 @@
 resource_prefix = app.node.try_get_context('resource_prefix') or 'dataall'
 
 source = app.node.try_get_context('repository_source') or 'codecommit'
+repo_string = app.node.try_get_context('repo_string') or 'awslabs/aws-dataall'
+repo_connection_arn = app.node.try_get_context('repo_connection_arn')
```
Comment (dlpzx) [inline deploy/app.py:73]: I am aware that it is not ensured for all the configuration options. But, we are trying to throw errors earlier for users when deploying data.all. Could you add a check here that if the source is `github` and `repo_connection_arn` is None it throws an exception?

File: deploy/app.py [hunk 2]
```diff
@@ -69,6 +69,11 @@
...
```
Comment (dlpzx) [inline deploy/app.py:73]: For future reference:  If the requested value is not defined,  app.node.try_get_context returns `None` in Python rather that raising an exception. So, the current implementation is correct.

File: deploy/app.py (hunk 1 above)
Comment (dlpzx) [reply deploy/app.py:73]: I have taken a closer look to the code and I think there is a better place to add the check for the github values. In `deploy.stacks.pipeline.PipelineStack.validate_deployment_params` we validate other params, we could also pass the source and connection_arn to this function

File: deploy/stacks/pipeline.py [hunk 3]
```diff
@@ -112,10 +116,10 @@ def __init__(
         self.pipeline_bucket.grant_read_write(iam.AccountPrincipal(self.account))
 
         if self.source == 'github':
```
Comment (dlpzx) [inline deploy/stacks/pipeline.py:118]: Can we rename this to 'codestar'? This will allow users to use GitLab and BitBucket right?

Comment (dlpzx) [reply deploy/stacks/pipeline.py]: Done, after agreeing offline with @asifma

**Comment by dlpzx** [comment]
@asifma is using CodeStar connections in AWS. In addition I tested that the CICD pipeline using CodeCommit is unnafected, that introducing a bad connection_arn throws an exception and that we can migrate from a CodeCommit deployment to a CodeStar connection deployment:

Error from bad input:
image

Successfully updated!
image

**Review by dlpzx — APPROVED** [review]

**Pull request merged**